src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -114,7 +114,11 @@
     <!-- Any child bound nodes that we need to preserve are put here. -->
     <Field Name="ChildBoundNodes" Type="ImmutableArray&lt;BoundNode&gt;"/>
   </Node>
-
+  
+  <Node Name="BoundTypeReferenceExpression" Base="BoundExpression">
+    <Field Name="SourceType" Type="Cci.ITypeReference" Null="disallow"/>
```

Comment:
It is strange to see "CCI" in the bound nodes. Feels like a layering issue. 
Generally CCI stuff comes up as an implementation detail at emit stage as a result of translating our symbols to the CCI model.

Why is this not a TypeSymbol?